### DISCUSSION TITLE
Add optional note to a transaction (visible to both sender and receiver)

### DISCUSSION THREAD

**Discussion by rafaeldjpbrochado** [iotaledger/firefly#1567] [Features / Enhancements]
## Description

I would like to suggest adding a `Note (optional)` string field for when users are trying to send funds. You know how in Paypal, Revolut and other banking apps when you try to send money they let you write a short optional note so that the other party knows what the money is for? That's basically what I'm requesting. Both the sender and the receiver should be able to see the note on a transaction, and ideally these notes should persist in the Tangle.

P.S.: I know Firefly will eventually have a communications module (akin to Signal, or Messenger) but this feature is different.

## Use cases

When sending money to a friend or family member, you can add a quick note saying what that money was for. E.g. "Happy birthday, mom!" or "This is for the pizza you paid for last night!"

It's also good even for your own bookkeeping. You can always go back through your transaction history and see what each transaction was for, assuming you attached notes to them before you sent them.

## Open questions

Is it possible to append a note string to an actual value transaction and store it all on the Tangle so that both the sender and receiver can see the note?

## Are you planning to do it yourself in a pull request?

No, but depending on how busy I am in the coming week(s), I might come back to this thread with some mock-ups for what this could look like in the app.

**Comment by Phyloiota** [comment]
like that idea! It could be just an additional payload to the value transfer and firefly should be able to recognize it and show it.

**Comment by Johnhersh** [comment]
Wondering if this is still being considered? I think it's a pretty essential feature. I'm trying to make an app with micro transactions, and so I want to collect them as dust in a single address. Being able to attach meaning to an incoming transaction would make things a lot easier.
